tests/gis_tests/test_geoip2.py

```diff
@@ -57,99 +90,101 @@ def test01_init(self):
-
-    @mock.patch("socket.gethostbyname")
-    def test03_country(self, gethostbyname):
-        "GeoIP country querying methods."
-        gethostbyname.return_value = "128.249.1.1"
-        g = GeoIP2(city="<foo>")
-
-        for query in (self.fqdn, self.addr):
-            self.assertEqual(
-                "US",
-                g.country_code(query),
-                "Failed for func country_code and query %s" % query,
-            )
-            self.assertEqual(
-                "United States",
-                g.country_name(query),
-                "Failed for func country_name and query %s" % query,
-            )
-            self.assertEqual(
-                {"country_code": "US", "country_name": "United States"},
-                g.country(query),
-            )
-
-    @mock.patch("socket.gethostbyname")
-    def test04_city(self, gethostbyname):
-        "GeoIP city querying methods."
-        gethostbyname.return_value = "129.237.192.1"
-        g = GeoIP2(country="<foo>")
-
-        for query in (self.fqdn, self.addr):
-            # Country queries should still work.
-            self.assertEqual(
-                "US",
-                g.country_code(query),
-                "Failed for func country_code and query %s" % query,
-            )
-            self.assertEqual(
-                "United States",
-                g.country_name(query),
-                "Failed for func country_name and query %s" % query,
-            )
-            self.assertEqual(
-                {"country_code": "US", "country_name": "United States"},
-                g.country(query),
-            )
-
-            # City information dictionary.
-            d = g.city(query)
-            self.assertEqual("NA", d["continent_code"])
-            self.assertEqual("North America", d["continent_name"])
-            self.assertEqual("US", d["country_code"])
-            self.assertEqual("Lawrence", d["city"])
-            self.assertEqual("KS", d["region"])
-            self.assertEqual("America/Chicago", d["time_zone"])
-            self.assertFalse(d["is_in_european_union"])
-            geom = g.geos(query)
-            self.assertIsInstance(geom, GEOSGeometry)
-
-            for e1, e2 in (
-                geom.tuple,
-                g.lon_lat(query),
-                g.lat_lon(query),
-            ):
-                self.assertIsInstance(e1, float)
-                self.assertIsInstance(e2, float)
-
-    def test06_ipv6_query(self):
-        "GeoIP can lookup IPv6 addresses."
-        g = GeoIP2()
-        d = g.city("2002:81ed:c9a5::81ed:c9a5")  # IPv6 address for www.nhm.ku.edu
-        self.assertEqual("US", d["country_code"])
-        self.assertEqual("Lawrence", d["city"])
-        self.assertEqual("KS", d["region"])
+    def test_bad_query(self):
+        g = GeoIP2(city="<invalid>")
+
+        functions = (g.city, g.geos, g.lat_lon, g.lon_lat)
+        values = (123, 123.45, b"", (), [], {}, set(), frozenset(), GeoIP2)
```

Comment:
I'd move `values` closer to where they are used:
```python
functions += (g.country, g.country_code, g.country_name)
values = (123, 123.45, b"", (), [], {}, set(), frozenset(), GeoIP2)
```

Reply:
Fair enough. Think I'd put it up higher to avoid redefining it over and over in the loop, but it's not expensive 👍🏻

Reply:
> Think I'd put it up higher to avoid redefining it over and over in the loop, but it's not expensive

But it's not in the loop :thinking:

Reply:
Forgive me... Pre-coffee and viewed the diff on a mobile browser with mega wrapping 😂